src/crypto/crypto_tls.cc

```diff
@@ -237,6 +238,13 @@ int SelectALPNCallback(
   if (UNLIKELY(alpn_buffer.IsEmpty()) || !alpn_buffer->IsArrayBufferView())
     return SSL_TLSEXT_ERR_NOACK;
 
+  bool alpn_fallback_allowed =
+      w->object()
+          ->Get(env->context(), env->allow_alpn_fallback_string())
+          .ToLocalChecked()
+          .As<Boolean>()
+          ->Value();
```

Comment:
As @bnoordhuis mentions, this is going to be slow. Definitely should avoid uses of `->Get(...)`

Reply:
I'm not very familiar with the JS/C++ interface. What's the preferred way to set a value from the JS TLS wrap such that it's directly accessible here?

If I create a new private field & setter on the C++ wrap, call that from JS in setup, and then read `w->allow_alpn_fallback` here instead of these calls, is that a) idiomatic and b) sufficiently performant?

Reply:
Yes and yes. :-)